edge/client/interface.go

```diff
@@ -87,11 +87,12 @@ type StackStatus struct {
 
 type setEndpointIDFn func(portainer.EndpointID)
 type getEndpointIDFn func() portainer.EndpointID
+type resetEdgeStacksFn func()
 
 // NewPortainerClient returns a pointer to a new PortainerClient instance
-func NewPortainerClient(serverAddress string, setEIDFn setEndpointIDFn, getEIDFn getEndpointIDFn, edgeID string, edgeAsyncMode bool, agentPlatform agent.ContainerPlatform, metaFields agent.EdgeMetaFields, httpClient *edgeHTTPClient) PortainerClient {
+func NewPortainerClient(serverAddress string, setEIDFn setEndpointIDFn, getEIDFn getEndpointIDFn, resetESFn resetEdgeStacksFn, edgeID string, edgeAsyncMode bool, agentPlatform agent.ContainerPlatform, metaFields agent.EdgeMetaFields, httpClient *edgeHTTPClient) PortainerClient {
```

Comment:
A slightly simpler approach could be to fix the inconsistency we have here: https://github.com/portainer/agent/blob/develop/edge/client/portainer_edge_async_client.go#L413:L424 so it matches this code: https://github.com/portainer/agent/blob/develop/edge/client/portainer_edge_client.go#L170:L179, then we can reset the endpoint ID (and stacks manager) directly in `poll_async.go` as `poll.go` does (https://github.com/portainer/agent/blob/develop/edge/poll.go#L217:L225). This way we avoid the need for a new callback.

Reply:
I agree. Initially, I was unsure if it would disrupt the flow since I assumed there must have been a reason for injecting the callback function into the client. It's great to have your confirmation, and I'll go ahead and simplify the flow a bit.